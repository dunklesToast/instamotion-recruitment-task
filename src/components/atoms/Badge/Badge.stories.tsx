import React from 'react';
import type { Meta, Story } from '@storybook/react';
import type { BadgeProps } from './Badge';
import { Badge } from './Badge';

export default {
  title: 'Atoms/Badge',
  component: Badge,
} as Meta;

const BadgeComponent: Story<BadgeProps> = (args) => <Badge {...args} />;

export const Default = BadgeComponent.bind({});
Default.args = {
  children: 'Story of Badge',
};
