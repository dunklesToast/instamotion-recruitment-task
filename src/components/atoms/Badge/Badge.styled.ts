import styled from 'styled-components';

export const Badge = styled.div`
  background-color: rgba(196, 196, 196, 0.5);
  border: #c4c4c4 1px solid;
  border-radius: 4px;
  display: inline-block;
  padding: 6px 8px;
`;
