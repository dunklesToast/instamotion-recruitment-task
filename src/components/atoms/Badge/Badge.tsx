import React from 'react';
import { Text } from '@atoms/Text/Text';
import * as S from './Badge.styled';

export interface BadgeProps {
  /** Text Content */
  children: React.ReactNode;
}

/**
 * Description of Badge.
 */
export function Badge({ children }: BadgeProps): JSX.Element {
  return (
    <S.Badge data-testid='badge'>
      <Text>{children}</Text>
    </S.Badge>
  );
}
