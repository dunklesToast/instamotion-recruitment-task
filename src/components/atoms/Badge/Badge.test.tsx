import React from 'react';
import { render, screen } from '@testing-library/react';
import { Badge } from './Badge';

test('Badge', () => {
  render(<Badge>Badge</Badge>);
  expect(screen.getByTestId('badge')).toBeInTheDocument();
  expect(screen.getByTestId('badge')).toMatchSnapshot();
});
