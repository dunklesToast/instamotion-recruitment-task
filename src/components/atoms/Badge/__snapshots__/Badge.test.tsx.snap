// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge 1`] = `
.c1 {
  font-family: 'PT Sans',sans-serif;
  font-style: regular;
  font-weight: 400;
}

.c0 {
  background-color: rgba(196,196,196,0.5);
  border: #c4c4c4 1px solid;
  border-radius: 4px;
  display: inline-block;
  padding: 6px 8px;
}

<div
  class="c0"
  data-testid="badge"
>
  <p
    class="c1"
  >
    Badge
  </p>
</div>
`;
